app/models/health/agency_performance.rb

```diff
@@ -247,5 +247,15 @@ def qualifying_activity_dates
         pluck(:patient_id, :date_of_activity).to_h
     end
 
+    def qa_signature_dates
+      @qa_signatures ||= Health::QualifyingActivity.submittable.
+        distinct.
+        where(patient_id: patient_referrals.keys). # limit to patients in scope
+        where(activity: :pctp_signed).
+        where(date_of_activity: @range).
+        order(date_of_activity: :asc).
+        pluck(:patient_id, :date_of_activity).to_h
```

Comment:
Should this be `group(:patient_id).maximum(:date_of_activity)`?

Reply:
Yep.